o2a/utils/el_utils.py

```diff
@@ -189,16 +189,35 @@ def comma_separated_string_to_list(line: str) -> Union[List[str], str]:
     return values[0] if len(values) <= 1 else values
 
 
-def normalize_path(url, props: PropertySet, allow_no_schema=False):
+def normalize_path(url: str, props: PropertySet, allow_no_schema=False):
+    """
+    Replaces all EL variables in the url, validates schema and returns only the 'path' part of a url.
+    Example: hdfs://localhost:8082/user/root --> user/root
+    """
+    url_with_var = replace_el_with_var(url, props=props, quote=False)
+    url_parts: ParseResult = urlparse(url_with_var)
+    if is_allowed_schema(allow_no_schema, url_with_var):
+        return url_parts.path
+
+
+def replace_url_el(url: str, props: PropertySet, allow_no_schema=False):
+    """
+    Replaces all EL variables in the url, validates schema and returns the url.
+    """
     url_with_var = replace_el_with_var(url, props=props, quote=False)
+    if is_allowed_schema(allow_no_schema, url_with_var):
+        return url_with_var
+
+
+def is_allowed_schema(allow_no_schema: bool, url_with_var: str):
     url_parts: ParseResult = urlparse(url_with_var)
     allowed_schema = {"hdfs", ""} if allow_no_schema else {"hdfs"}
     if url_parts.scheme not in allowed_schema:
         raise ParseException(
             f"Unknown path format. The URL should be provided in the following format: "
             f"hdfs://localhost:9200/path. Current value: {url_with_var}"
         )
```

Comment:
We should return False here and throw exception in else above (is_allowed_schema suggests True/False response). Also adding -> bool in signature would be great.

Reply:
Do you mean sth like this?
```
    if url_parts.scheme in allowed_schema:
        return True
    else:
        raise ParseException(
            f"Unknown path format. The URL should be provided in the following format: "
            f"hdfs://localhost:9200/path. Current value: {url_with_var}"
        )
```

I guess we should either return `False` or raise an exception, right? Doing both in the same clause makes little sense because the second statement will never get invoked (if return is first, the exception will never be raised, and vice versa).

Reply:
Actually after running pre-commit on all files (somehow these files were not checked at my local machine before...) it pointed me to some problems in this code, which I addressed in 19e9da4766727116b15c1fb04adb2836b797c2d1.
Feel free to take a look, maybe it will resolve this issue :)
There's some duplication as to throwing exceptions but probably it's a bit clearer overall.

Reply:
That's exactly what I meant :).